js/qmwp-bargraph-gauge-scatterplot-timeline_charts.js

```diff
@@ -192,66 +192,157 @@ AnalyzeChart = function () {
             AnalyzePage.hideCorrelationGauge();
         }
         else {
-            AnalyzePage.showScatterplot();
-            Quantimodo.getPairs({
-                    'effect': effect.originalName,
-                    'cause': cause.originalName,
-                    'startTime': AnalyzePage.getStartTime(),
-                    'endTime': AnalyzePage.getEndTime()
-                },
-                function (measurements) {
-                    if (measurements.length) {
-                        var scatterplotDots = [];
-                        var xMax = -Infinity, yMax = -Infinity, xMin = +Infinity, yMin = +Infinity;
+            //use effect and cause to create pairs
 
-                        for (var i in measurements) {
-                            var dot = {
-                                x: measurements[i].causeMeasurement,
-                                y: measurements[i].effectMeasurement
-                            };
+            function getValueForDate(measurements, date) {
```

Comment:
Function declarations should not be placed in blocks. Use a function expression or move the statement to the top of the outer function.